src/Microsoft.OData.Core/ODataBatchReader.cs

```diff
@@ -204,440 +236,81 @@ void IODataBatchOperationListener.BatchOperationContentStreamDisposed()
         }
 
         /// <summary>
-        /// Returns the next state of the batch reader after an end boundary has been found.
+        /// Increases the size of the current batch message; throws if the allowed limit is exceeded.
         /// </summary>
-        /// <returns>The next state of the batch reader.</returns>
-        private ODataBatchReaderState GetEndBoundaryState()

```

Comment:
>GetEndBoundaryState [](start = 38, length = 19)

I would expect these states and transitions to be common between the multipart/mixed and JSON formats, in which case they should be in the base class.